src/value/mod.rs

```diff
@@ -838,6 +838,395 @@ mod generic_value_tests {
+                            E::Builder::new_string("bye"),
+                        ),
+                    ]
+                    .into_iter(),
+                ),
+            ],
+        }
+    }
+
+    fn struct_with_duplicates_in_multiple_fields_case<E: Element>() -> CaseStruct<E>
+    where
+        E::Builder: Builder<SymbolToken = E::SymbolToken>,
+    {
+        CaseStruct {
+            eq_annotations: vec![
+                E::Builder::new_struct(
+                    vec![
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                    ]
+                    .into_iter(),
+                ),
+                // structs with different symbol token sids
+                E::Builder::new_struct(
+                    vec![
+                        (
+                            E::SymbolToken::text_token("a").with_local_sid(21),
+                            E::Builder::new_i64(1),
+                        ),
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                    ]
+                    .into_iter(),
+                ),
+            ],
+            ne_annotations: vec![
+                // structs with different length
+                E::Builder::new_struct(
+                    vec![
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                    ]
+                    .into_iter(),
+                ),
+                // structs with annotated values
+                E::Builder::new_struct(
+                    vec![
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                        (
+                            E::SymbolToken::text_token("a"),
+                            E::Builder::new_i64(1)
+                                .with_annotations(vec![E::SymbolToken::text_token("a")]),
+                        ),
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                    ]
+                    .into_iter(),
+                ),
+                // structs with different value for duplicates
+                E::Builder::new_struct(
+                    vec![
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(2)),
+                        (E::SymbolToken::text_token("a"), E::Builder::new_i64(1)),
+                    ]
+                    .into_iter(),
+                ),
+            ],
+        }
+    }
+
+    fn struct_with_duplicate_fieldnames_case<E: Element>() -> CaseStruct<E>
+    where
+        E::Builder: Builder<SymbolToken = E::SymbolToken>,
```

Comment:
What about a variant(s) of these that have sources and/or no text either alone or mixed with text?